include/grpcpp/alts_context.h

```diff
@@ -0,0 +1,76 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPCPP_ALTS_CONTEXT_H
+#define GRPCPP_ALTS_CONTEXT_H
+
+#include <grpc/grpc_security_constants.h>
+#include <grpcpp/impl/codegen/security/auth_context.h>
+
+#include <memory>
+
+#include "src/proto/grpc/gcp/altscontext.upb.h"
```

Comment:
@yang-g 
Hi Yang, this PR is almost ready for you to review and I will let members in our team to review this first.
Before sending you for reviewing, can I please ask you a question? I want to forward declare grpc_gcp_AltsContext and move this include to the .cc file. However, since this grpc_gcp_AltsContext is a C struct instead of a smart pointer, it wouldn't be able to compile if I just declare it in the header file and use it in source file. May I know typically in C++, are there any ways to solve this problem?

Reply:
I would imagine it should work. Does it fail to compile and what is the error message if you try the forward declaration?

Reply:
I just tried and it seemed to work with the change...

Reply:
The diff:
```
--- a/include/grpcpp/alts_context.h
+++ b/include/grpcpp/alts_context.h
@@ -24,7 +24,7 @@
 
 #include <memory>
 
-#include "src/proto/grpc/gcp/altscontext.upb.h"
+struct grpc_gcp_AltsContext;
 
 namespace grpc {
 
diff --git a/src/cpp/common/alts_context.cc b/src/cpp/common/alts_context.cc
index b39672140b..28bc656353 100644
--- a/src/cpp/common/alts_context.cc
+++ b/src/cpp/common/alts_context.cc
@@ -22,6 +22,7 @@
 #include "src/core/lib/gprpp/memory.h"
 #include "src/core/tsi/alts/handshaker/alts_tsi_handshaker.h"
 #include "src/cpp/common/secure_auth_context.h"
+#include "src/proto/grpc/gcp/altscontext.upb.h"
 
 namespace grpc {

```

Reply:
Thanks for helping me debugging this! It turns out that I wrongly declare the struct inside the grpc namespace:(
@yang-g Could you please take a look at the PR now? I will cherry pick it to internal to make sure the build goes well after your review. Thanks again for the help!